L1Trigger/L1TMuon/src/GeometryTranslator.cc

```diff
@@ -134,11 +136,31 @@ void GeometryTranslator::checkAndUpdateGeometry(const edm::EventSetup& es) {
 
 GlobalPoint
 GeometryTranslator::getGEMSpecificPoint(const TriggerPrimitive& tp) const {
-  const GEMDetId id(tp.detId<GEMDetId>());
-  const GEMEtaPartition * roll = _geogem->etaPartition(id);
-  const uint16_t pad = tp.getGEMData().pad;
-  const LocalPoint lp = roll->centreOfPad(pad);
-  const GlobalPoint gp = roll->toGlobal(lp);
+  LocalPoint lp;
+  GlobalPoint gp;
+
+  if (!tp.getGEMData().isME0) {  // use GEM geometry
+    const GEMDetId id(tp.detId<GEMDetId>());
+    const GEMEtaPartition * roll = _geogem->etaPartition(id);
+    assert(roll);
+    //const uint16_t pad = tp.getGEMData().pad;
+    // Use half-strip precision, - 0.5 at the end to get the center of the strip
+    const float pad = (0.5 * static_cast<float>(tp.getGEMData().pad_low + tp.getGEMData().pad_hi)) - 0.5;
+    lp = roll->centreOfPad(pad);
+    gp = roll->surface().toGlobal(lp);
+
+  } else {  // use ME0 geometry
```

Comment:
Is ME0 part of the GEMs? Or are they sharing a data format or something?  If ME0 and GEM will have distinct data formats / etc., I suspect there should be a separate getME0SpecificPoint ... but you know more about this than I do.